### PR TITLE
Allow PHPMD to run on PHP 8.0

### PR DESCRIPTION
**Pull request by wandersonwhcr**
Type: feature
Breaking change: no

This PR updates Composer PHP requirement to 8.0 and fixes warnings when using this version.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ matrix:
       env: BUILD_PHAR=true
     - php: 7.3
       env: WEBSITE=true
+    - php: 7.4
+    - php: 7.4
+      env: DEPENDENCIES=low
```
Comment (kylekatarnls) [inline .travis.yml:17]:
```suggestion
    - php: 8.0
```
We would except tests to also pass on 8.0

(not yet with DEPENDENCIES=low before we bump PDepend version to 2.9.0 which can run on PHP 8)

Comment (wandersonwhcr) [reply .travis.yml:17]: Yes, after I created this PR I found a `.travis.yml` file.

Next hours I will fix problems with dependencies, too.

**Comment by tvbeek** [comment]
I see this PR targets the 3-x branch, but we currently working on the 2.x versions. That means that this PR will not be very soon part of a release. (It is still welcome but I want you to know it before we get some misunderstanding)

**Comment by wandersonwhcr** [comment]
> I see this PR targets the 3-x branch, but we currently working on the 2.x versions. That means that this PR will not be very soon part of a release. (It is still welcome but I want you to know it before we get some misunderstanding)

When I tried to use 2.8.x branch, I found that PHPUnit version was not compatible with PHP 8.0. I created a local branch and migrated the source to latest version of PHPUnit.

After, I think: anyone did that too? So, I found 3.x branch, where PHPUnit was migrated to 8.0. This is why I use this branch :smile: 

But this is not the current problem, this is the first step to another problem where I can't define anonymous classes and use PHPMD to scan it. I will describe my problem in the next hours.

**Comment by tvbeek** [comment]
> > I see this PR targets the 3-x branch, but we currently working on the 2.x versions. That means that this PR will not be very soon part of a release. (It is still welcome but I want you to know it before we get some misunderstanding)
> 
> When I tried to use 2.8.x branch, I found that PHPUnit version was not compatible with PHP 8.0. I created a local branch and migrated the source to latest version of PHPUnit.
> 
> After, I think: anyone did that too? So, I found 3.x branch, where PHPUnit was migrated to 8.0. This is why I used this branch :smile: 


You could use the master branch, that is the active branch (I need to verify why we have a 2.8.x branch)  
Currently we support a lot for PHP 8 but we have some open issues for new functions.

**Comment by kylekatarnls** [comment]
@wandersonwhcr first of all, PDepend which is used by PHPMD to parse the code can run on PHP 8 only since the version 2.9.0 and the syntax support should be complete soon with the version 2.9.1. If you're using the stable-release phar version, you're likely using 2.8.0 which would be the blocking point here.

If so you can use the last phar generated on our master branch:
https://github.com/phpmd/phpmd/suites/2487849749/artifacts/53642867

If you installed PHPMD with composer, a simple `composer update` should help to update `pdepend/pdepend`.

Next phar release of PHPMD (2.10.0) will embed PDepend 2.9.1 and should support PHP 8 with no need for any further change in the code or the composer.json.

**Comment by wandersonwhcr** [comment]
Hello,

Yesterday I tried to reproduce the error using this repository inside unit tests and I could not. I created a test case and it passed.

So, I created a repository to reproduce the error.

https://github.com/wandersonwhcr/phpmd-with-errors

Yes, like @kylekatarnls described, I think it is a PDepend error. But I compared the versions from required packages and they are the same on both repositories, including PDepend on version `2.9.0`. Some Symfony packages differ, but I think they are dependencies from `phpmd` CLI command.

The only real difference I see is on my repository with errors, I use `dist` package and when I try to reproduce the error on PHPMD repository, it is a `source` package.

PS: I tried to reproduce the error on tag `2.9.1` from PHPMD repo.

I am lost here, sorry :cry:

**Comment by wandersonwhcr** [comment]
Done.

I know... My PR is more complicated than #878 and I didn't know that @kylekatarnls was trying to update to PHP 8.0.

#878 is an elegant solution.

But, if you want an approach that tries to upgrade PHPUnit, this PR is the solution.

If you don't, please, can you close it?

Thank you <3

**Comment by kylekatarnls** [comment]
Hello my PR is just the continuation to integrate the new PDepend version wich supports PHP 8 but for PHPMD 3 we'll clearly upgrade phpunit.

**Comment by vpassapera** [comment]
Just FYI, not sure if explicitly known, but the biggest issue I'm having is with named parameters and piped return values.

**Comment by wandersonwhcr** [comment]
too old. closing.

**Pull request closed without merging**